### PR TITLE
add private network to network enum

### PR DESCRIPTION
**Pull request by error2215**

### CODE REVIEW

File: corebc-core/src/types/network.rs [hunk 1]
```diff
@@ -139,7 +140,45 @@ impl Serialize for Network {
...
+                        if let Ok(id) = id.parse::<u64>() {
+                            return Network::Private(id)
+                        }
+                    }
+                }
+                panic!("Unknown network: {}", unknown);
+            }
+        }
+    }
+}
```
Comment (Kuly14) [inline corebc-core/src/types/network.rs]:
```suggestion
 fn from(s: String) -> Network {
    match s.as_str() {
        "mainnet" => Network::Mainnet,
        "devin" => Network::Devin,
        unknown => {
            if let ["private", id_str] = unknown.split('-').collect::<Vec<_>>().as_slice() {
                if let Ok(id) = id_str.parse::<u64>() {
                    return Network::Private(id);
                }
            }
            panic!("Unknown network: {}", unknown);
        }
    }
}
```

This makes it a bit more readable, so we don't have 3 nested if lets.

File: corebc-core/src/types/network.rs [hunk 2]
```diff
@@ -1,36 +1,18 @@
...
-//     - Serde (in snake_case): `#[serde(alias = "<aliasX>", ...)]`
-//      Aliases are appended to the `Deserialize` implementation.
-//      More info: <https://serde.rs/variant-attrs.html>
 //     - Add a test at the bottom of the file
 
-// We don't derive Serialize because it is manually implemented using AsRef<str> and it would
-// break a lot of things since Serialize is `kebab-case` vs Deserialize `snake_case`.
-// This means that the Network type is not "round-trippable", because the Serialize and Deserialize
-// implementations do not use the same case style.
-
```
Comment (Kuly14) [inline corebc-core/src/types/network.rs:33]: Did you find out why is this the case? It doesn't really make sense to me to make the network type not "round-trippable"

**Review by Kuly14 — COMMENTED** [review]
Other than the 1 suggestion I think it looks great.

File: corebc-core/src/types/network.rs (hunk 2 above)
Comment (error2215) [reply corebc-core/src/types/network.rs:33]: not sure about that, but thats what I found about this 
https://github.com/gakonst/ethers-rs/issues/2038#issuecomment-1377773079
https://github.com/gakonst/ethers-rs/pull/2268#discussion_r1138931364

https://github.com/gakonst/ethers-rs/pull/2270/files
Seems like it was hot-fixed and tests for "round-trippable" were added

**Comment by error2215** [comment]
Also I removed utils::NetworkType and replaced it with types::Network. The same logic was written twice.
Tests pass
`cargo +nightly fmt --all` and `cargo +nightly clippy` applied

**Review by Kuly14 — APPROVED** [review]

**Pull request merged**